src/index.tsx

```diff
@@ -209,10 +211,20 @@ export default class ReactToPrint extends React.Component<IReactToPrintProps> {
     public render() {
         const {
             trigger,
+            onBeforeGetContent
         } = this.props;
 
         return React.cloneElement(trigger(), {
-            onClick: this.handlePrint,
+            onClick: () => {
+                const onBeforeGetContentPromise = onBeforeGetContent();
+                if(onBeforeGetContentPromise && typeof onBeforeGetContentPromise.then === "function") {
+                    onBeforeGetContentPromise.then(() => {
+                    return this.handlePrint();
```

Comment:
The `return` here isn't needed - `this.handlePrint()` does not return anything

Reply:
I need to return the `this.handlePrint()` since all function is inside a **{}**
if nothing is returned, the function won't be called by React on onClick